src/concepts/plugins/plugin_basics.md

```diff
@@ -0,0 +1,430 @@
+
+The `.gitignore`, `LICENSE` and `README.md` files are excluded as they are ultimately immaterial to the Plugin developement process from an end-user perspective.
+:::
+
+#### public
+
+The `public` directory stores the `styles.css` file used to stylize elements of your Plugin.
+
+#### src
+
+The `src` directory stores the following files:
+
+- `index.ts`: This file contains the main code of your Plugin. _View the [EntryPoint File](#the-entrypoint-file) section for more information._
+- `types.ts`: This file defines and exports type delcarations/definitions to be bundled with your code and be consumable by dependents.
+
+#### Package Management
+
+Internally, Caido utilizes the [Pinned Node Package Manager(a.k.a. pnpm)](https://pnpm.io/) for Plugin package management. In general, a package manager handles the installation, updating and removal of packages. This includes ensuring the correct dependency versions are installed and resolving and dependency conflicts between packages.
+
+The files related to package management within the StarterKit repository are:
+
+- `package.json`: This is the main configuration file for the package as a whole - it contains the project metadata and specifies the required dependencies. _View the [pnpm package.json](https://pnpm.io/package_json) documentation for more information._
+- `pnpm-lock.yaml`: This file ensures the same versions of dependencies are installed with every installation. _View the [pnpm-lock.yaml](https://pnpm.io/git#lockfiles) documentation for more information._
+
+#### TypeScript
+
+Externally, [TypeScript](https://docs.caido.io/concepts/essentials/workflows/js_in_caido.html#typing) is used by Caido for the StarterKit package.
+
+The file related to TypeScript within the StarterKit repository is:
+
+- `tsconfig.json`: This file provides instructions to the compiler when TypeScript is converted to JS. _View the [https://www.typescriptlang.org/docs/handbook/tsconfig-json.html](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) documentation for more information._
+
+#### Build Tool
+
+Once the package is developed, the package is processed by the [Vite](https://vitejs.dev/guide/) build tool. In general, a build tool automates the process of compiling, testing and packaging code into a deployable package - ensuring the Plugin is ready for use and sharing.
+
+The file related to the Vite build tool within the StarterKit repository is:
+
+- `vite.config.ts`: This file is a configuration file for customizing the build process. _View the [vite.config.js](https://v2.vitejs.dev/config/) documentation for more information.
+
+#### Plugin Specific Configuration
+
+The `manifest.json` file is specific to the Plugin itself and defines the metadata and configurations. This file also represents the structure of the Plugin.
```

Comment:
I want us to make the distinction between plugin and plugin package. Manifest is for the package

Reply:
So I got them mixed up: package.json is for the plugin specifically and manifest.json is for the whole package?